numpy/core/src/multiarray/ctors.c

```diff
@@ -1816,11 +1816,9 @@ PyArray_FromAny(PyObject *op, PyArray_Descr *newtype, int min_depth,
     }
 
     /* If the requested dtype is flexible, adapt it */
-    if (newtype != NULL) {
-        PyArray_AdaptFlexibleDType(op,
+    newtype = PyArray_AdaptFlexibleDType(op,
                     (dtype == NULL) ? PyArray_DESCR(arr) : dtype,
-                    &newtype);
-    }
+                    newtype);
```

Comment:
There is no need for an error check here?

Reply:
yes, fixed